### ISSUE TITLE
Outgoing HTTP connection & API Keys

### ISSUE THREAD

**Issue by farefernandez** [zatosource/zato#398]
I'm trying to issue a request against an external HTTP/JSON API which relies upon a custom header to perform authentication (X-API-Key).

Even though I'm able to define such authentication mechanism under Security -> Api Keys in the web admin, I'm not able to pick it from the outgoing channel definition. It shows for instance HTTP basic auth credentials I had created for another connection, but not the ones based on Api Keys.

As for the discussion on the mailing list, this can't be done via the web ui in Zato 2.0. To obtain the same result, custom code must be used:

```
conn = self.outgoing.plain_http['My Conn'].conn
auth = {'X-API-Key': 'My Key'}
response = conn.post(self.cid, 'Request', headers=auth)
```

I'm proposing/suggesting to make the Api Keys auth mechanism available to outgoing HTTP channels.

**Comment by jcmcken** [comment]
Any concrete plans for this change in 2.x or 3? This and a few of the other security providers would be really handy to configure against outgoing HTTP connections directly.

**Comment by dsuch** [comment]
Hi @jcmcken,

this can happen for Python 3.0 and it's not a huge amount of work but it needs a sponsoring company to make sure it's included.

After 3.0, Zato 2.0.x will be put in maintenance and no features will be added, only fixes.

What other security providers are you thinking of? Feel free to contact me directly on dsuch@zato.io if you'd like to discuss any details.

Thanks!

**Comment by dsuch** [comment]
I'm setting the Milestone to 3.0 and assigning to myself.

@jcmcken - you said there were other security features that you'd like to see. Can you contact me on dsuch@zato.io if you are interested in sponsoring them or otherwise describe what you were thinking of?

Regards.

**Comment by dsuch** [comment]
Done - will be released in 3.0 in a few weeks now.
